frappe/automation/doctype/assignment_rule/assignment_rule.py

```diff
@@ -47,19 +50,20 @@ def do_assignment(self, doc):
 		# clear existing assignment, to reassign
 		assign_to.clear(doc.get('doctype'), doc.get('name'))
 
-		user = self.get_user()
+		users = self.get_users()
 
-		assign_to.add(dict(
-			assign_to = user,
-			doctype = doc.get('doctype'),
-			name = doc.get('name'),
-			description = frappe.render_template(self.description, doc),
-			assignment_rule = self.name,
-			notify = True
-		))
+		for user in users:
+			assign_to.add(dict(
+				assign_to = user,
+				doctype = doc.get('doctype'),
+				name = doc.get('name'),
+				description = frappe.render_template(self.description, doc),
+				assignment_rule = self.name,
+				notify = True
+			))
 
-		# set for reference in round robin
-		self.db_set('last_user', user)
+			# set for reference in round robin
+			self.db_set('last_user', user)
```

Comment:
This solves the problem, but, this will cause the `last_user` to be set multiple times which is not efficient as we're doing unnecessary database writes. 

Instead do `self.db_set('last_user', users[-1])` or in fact `self.db_set('last_user', user)` will also work as the `user` variable is set to the last element in that list, but it doesn't read well.